spec/ruby/floating_points_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'floating_points'
+
+RSpec.describe FloatingPoints do
+  it 'parses test properly' do
+    r = FloatingPoints.from_file('src/floating_points.bin')
+    
+    delta = 1e-6
+
+    expect(r.single_value).to be_within(delta).of 1.2345
+    expect(r.single_value_be).to be_within(delta).of 1.2345
+    expect(r.double_value).to be_within(delta).of 123.456
+    expect(r.double_value_be).to be_within(delta).of 123.456
```

Comment:
Actually, it doesn't have much with exactly Ruby. It's how numbers are stored internally, there is actually no way to store exact value of stuff like 1.2345 in binary numeric system, so everyone will resort to using an approximation:

## Ruby

```ruby
def pack_and_unpack(x); [x].pack('f').unpack('f')[0]; end

pack_and_unpack(1) # => 1.0
pack_and_unpack(1.25) # => 1.25
pack_and_unpack(1.26) # => 1.2599999904632568
pack_and_unpack(1.234) # => 1.2339999675750732
pack_and_unpack(1.2345) # => 1.2345000505447388
```

## Python

```python
def pack_and_unpack(x): return unpack('f', pack('f', x))[0]

pack_and_unpack(1) # => 1.0
pack_and_unpack(1.25) # => 1.25
pack_and_unpack(1.26) # => 1.2599999904632568
pack_and_unpack(1.234) # => 1.2339999675750732
pack_and_unpack(1.2345) # => 1.2345000505447388
```

For both languages, 1.2345000505447388 != 1.2345. Actually, there is no such thing as precise "1.2345" anyway - it's just language's fancy rendition of infinite periodic non-integer value that represents it:

1.0011(1100000010000011000100100110111010010111100011010100111111011111001110110110010001011010000111001010)...

Both Ruby & Python internal types use double for floating point, thus dumping an into single precision and then converting them (internally) to double zeroes out some valuable mantissa bits, and thus this "fancy rendition" magic fails to work. I believe that would be the same with C, C#, Java, etc, unless you dump & retrieve it in exactly the same precisions.

You might want to choose floats that have exact, non-periodic rendition in binary for these tests. And maybe throw in one with the infinite period in, but clearly marked as so ;)